doc/tutorial/argument_parser/index.md

```diff
@@ -0,0 +1,414 @@
+3. Try calling `--help` again and see the results.
+
+\endassignment
+\solution
+
+\include doc/tutorial/argument_parser/solution1.cpp
+
+\endsolution
+
+# Adding options, flags and positional_options
+
+Now that we're done meta information, we will learn how to add the actual functionality of options, flags and positional options. For each of these three there is a respective member function:
+
+* seqan3::argument_parser::add_option
+* seqan3::argument_parser::add_flag
+* seqan3::argument_parser::add_positional_option
+
+All of the functions above take as the first parameter a variable by reference, which will directly store the corresponding parsed value from the command line. This has two advantages compared to other command line parsers: (1) There is no need for a getter function after parsing and (2) The type is automatically deduced (e.g. with boost::program_options you would need to access `parser["file_path"].as<std::filesystem::path>()` afterwards).
+
+The seqan3::argument_parser::add_flag only allows boolean variables while seqan3::argument_parser::add_option and seqan3::argument_parser::add_positional_option allow **any type that is convertible from a std::string via std::from_chars** or a container of the former (see \ref tutorial_argument_parser_list_options). Besides accepting generic types, the parser will **automatically check if the given command line argument can be converted into the desired type**, and if not throw a seqan3::type_conversion_failed exception.
+
+So how does this look like? The following code snippet adds a positional option to `parser`.
+
+\snippet doc/tutorial/argument_parser/small_snippets.cpp add_positional_option
+
+Additionally to the variable that will store the value, you need to pass a description. This description will help users of your application to understand how the option is affecting your program.
+
+\note As the name suggest, positional options are identified by their position. In SeqAn3, the first `add_positional_option()` will be linked to the first command line argument that is neither an option-value pair nor a flag. So the order of initialising your parser determines the order of assigning command line arguments to the respective variables.
+We personally recommend to always use regular options (id-value pairs) because they are more expressive and it is easier to spot errors.
+
+You can add an option like this:
+
+\snippet doc/tutorial/argument_parser/small_snippets.cpp add_option
+
+Additionally to the variable that will store the value and the description, you need to specify a short and long identifier. The example above will recognize an option `-n` or `--my-number` given on the command line and expect it to be followed by a value separated only by e.g. `=` or a space (see [this post](https://www.math.uni-hamburg.de/doc/java/tutorial/essential/attributes/_posix.html) for details).
```

Comment:
e.g. `=` or a space? are there more? the link doesn't seem to say something about it...